test/OrchardCore.Tests/Apis/ContentManagement/DeploymentPlans/ContentStepLuceneQueryTests.cs

```diff
@@ -50,5 +51,88 @@ public async Task ShouldUpdateLuceneIndexesOnImport()
             Assert.Equal("new version", nodes[0]["displayText"].ToString());
             Assert.Equal("second content item display text", nodes[1]["displayText"].ToString());
         }
+
+
+        [Fact]
+        public async Task ShouldLuceneIndexOnlyIndexesTheLatestversion()
+        {
+            using var context = new BlogPostDeploymentContext();
+            await context.InitializeAsync();
+            // Act
+            var recipe = BlogPostDeploymentContext.GetContentStepRecipe(context.OriginalBlogPost, jItem =>
+            {
+                jItem[nameof(ContentItem.ContentItemVersionId)] = "newVersion";
+                jItem[nameof(ContentItem.DisplayText)] = "new version";
+            });
+            const string secondcontentitemversionid = "secondcontentitemversionid";
+            var data = recipe["steps"][0]["Data"].AsArray();
+            var secondContentItem = JObject.FromObject(context.OriginalBlogPost);
+            secondContentItem[nameof(ContentItem.ContentItemVersionId)] = secondcontentitemversionid;
+            secondContentItem[nameof(ContentItem.DisplayText)] = "second content item version display text";
+            secondContentItem[nameof(AutoroutePart)][nameof(AutoroutePart.Path)] = "new-path";
+            data.Add(secondContentItem);
+
+            Assert.Equal(2, data.Count);
+            Assert.Single(data.Select(x => x.SelectNode(nameof(ContentItem.ContentItemId)).ToString()).Distinct());
+
+            // validate different contentItem versionId
+            var versionIds = data.Select(x => x.SelectNode(nameof(ContentItem.ContentItemVersionId)).ToString()).Distinct();
+            Assert.Equal(2, versionIds.Count());
+
+            await context.PostRecipeAsync(recipe);
+
+            // Search indexes are no longer updated in a deferred task at the end of a shell scope
+            // but in a background job after the http request, so they are not already up to date.
```

Comment:
Do we not want to test the DefaultContentManager instead then?

Reply:
Because the `DefaultContentManager` api always returns only the latest version

Reply:
Yes but we want to test the IndexingHandlers that are triggered by events from the DefaultContentManager. These indexing handlers should take care of indexing proper content item version.

Reply:
I'm confused, you mean we need to use `IContentManager` in this unit test to recheck whether the content items have been updated to the latest, right? Or other tests are needed

Reply:
Yes, I think what we want to test is what happens when a content items is created/drafted/updated with a user action with the IContentManager. Else, we are in the context of executing a background task. We don't have any unit tests for this either so what you did is not lost.